app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java

```diff
@@ -150,9 +157,51 @@ private void performOperation(int index, int position, File file) {
                 mPDFEncryptionUtils.setPassword(file.getPath(), ViewFilesAdapter.this, mFileList);
                 break;
 
-            case 7://Password  Remove
+            case 7://Password Remove
                 mPDFEncryptionUtils.removePassword(file.getPath(), ViewFilesAdapter.this, mFileList);
                 break;
+
+            case 8://Rotate Pages
+                new MaterialDialog.Builder(mActivity)
+                        .title(R.string.rotate_pages)
+                        .content(R.string.enter_rotation_angle)
+                        .input(null, null, false, (dialog, input) -> {
+                            String sourceFilePath = file.getPath();
+                            String destFilePath = sourceFilePath.substring(0, sourceFilePath.lastIndexOf('/') + 1);
+                            String fileName = sourceFilePath.substring(sourceFilePath.lastIndexOf('/') + 1);
+                            destFilePath += String.format(mActivity.getString(R.string.rotated_file_name),
+                                    fileName.substring(0, fileName.lastIndexOf('.')), input,
+                                    mActivity.getString(R.string.pdf_ext));
+                            try {
```

Comment:
Add this function in `PDFUtils.java`

Reply:
Whole code or just rotation code?

Reply:
just the rotation code.

Just send source & destination filename to that function.